raws-feed-processor/src/main/java/edu/mit/ll/nics/processor/raws/RAWSFeedProcessor.java

```diff
@@ -0,0 +1,125 @@
+            SimpleFeatureSource featureSource = datastore.getFeatureSource(rawsFeatureSource);
+            if (!(featureSource instanceof SimpleFeatureStore)) {
+                logger.error(String.format("Write denied on feature source : %s", featureSource));
+                return;
+            }
+
+            SimpleFeatureStore featureStore = (SimpleFeatureStore) featureSource;
+            transaction = new DefaultTransaction("raws-feed-tx");
+            featureStore.setTransaction(transaction);
+            SimpleFeatureType rawsFeatureType = featureStore.getSchema();
+            SimpleFeatureBuilder featureBuilder = new SimpleFeatureBuilder(rawsFeatureType);
+            for(RAWSFeature rawsFeature : rawsFeatures) {
+                if(this.updateFeature(rawsFeature, featureBuilder, featureStore, transaction, rawsFeatureType))
+                    successfulFeatureUpdates++;
+                else
+                    failedFeatureUpdates++;
+                if((successfulFeatureUpdates+failedFeatureUpdates)%100 == 0)
+                    logger.debug(String.format("So far processed %d features", (successfulFeatureUpdates+failedFeatureUpdates)));
+            }
+        } catch(Exception e) {
+            logger.error(String.format("Failed to process RAWS features successfully, ran for %d ms before failing", stopwatch.elapsed(TimeUnit.MILLISECONDS)), e);
+            if(transaction != null)
+                transaction.rollback();
+        } finally {
+            if(transaction != null)
+                transaction.close();
+        }
+        logger.info(String.format("Successfully completed processing %d RAWS Features & Failed to process %d RAWS Features in %d ms", successfulFeatureUpdates, failedFeatureUpdates, stopwatch.elapsed(TimeUnit.MILLISECONDS)));
+    }
+
+    public boolean updateFeature(RAWSFeature rawsFeature, SimpleFeatureBuilder featureBuilder, SimpleFeatureStore featureStore, DefaultTransaction transaction, SimpleFeatureType rawsFeatureType) throws Exception {
+        try {
+            Filter filter = CQL.toFilter("station_id = '" + rawsFeature.getRawsObservations().getStationId() + "'");
+            featureStore.removeFeatures(filter);
+            if(rawsFeature.getRawsObservations().getStatus().equalsIgnoreCase("ACTIVE")) { //add feature only if station status is active
+                addFeature(rawsFeature, featureBuilder, featureStore, rawsFeatureType);
+            }
+            transaction.commit();
+            return true;
+        } catch(RuntimeException e) {
+            throw e;
+        } catch(Exception e) {
+            logger.error(String.format("Failed to persist updates for %s, Error Details: ", rawsFeature.getRawsObservations()), e);
+            transaction.rollback();
+            return false;
+        }
+    }
+
+    public void addFeature(RAWSFeature rawsFeature, SimpleFeatureBuilder featureBuilder, SimpleFeatureStore featureStore, SimpleFeatureType rawsFeatureType) throws Exception {
```

Comment:
The naming between addFeature() and updateFeature() is a bit unclear, I think because the level of abstraction is so different.